### PR TITLE
feat(project): list  project and add project with back-end

### PR DESCRIPTION
**Pull request by qiuxiaolian**
**What this PR does / why we need it**:

screen shot 2019-01-24 at 6 41 38 pm

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @caosm @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cd1989** [comment]
@qiuxiaolian One suggestion: make the operation column right aligned.

File: web/src/components/public/resourceAllocation/Allocation.js [hunk 1]
```diff
@@ -15,26 +15,26 @@ const allocationMap = [
   {
```
Comment (cd1989) [inline web/src/components/public/resourceAllocation/Allocation.js]: `web/src/components/public/resourceAllocation/Allocation.js`

In backend the term is `quota`, not `allocation`.

File: web/src/less/inc.var.less [hunk 2]
```diff
@@ -54,6 +54,25 @@ code {
   justify-content: space-between;
 }
 
+.cyclone-layout-header {
```
Comment (caosm) [inline web/src/less/inc.var.less]: duplicate

File: web/src/components/public/quota/Quota.js [hunk 3]
```diff
@@ -150,67 +182,107 @@ class Allocation extends React.Component {
...
-                      render={props => (
-                        <InputWithUnit
-                          label={intl.get('allocation.cpuRequest')}
-                          addonAfter="Core"
-                          defaultAddon="Core"
-                          className="cpu"
-                          {...props}
-                        />
-                      )}
+                      render={props => {
```
Comment (caosm) [inline web/src/components/public/quota/Quota.js]: componentize is better

File: web/src/components/project/component/addProject/Form.js [hunk 4]
```diff
@@ -1,74 +1,111 @@
...
+          const services = _.map(values.spec.services, n => {
+            const resources = n.split('/');
+            return { type: resources[0], name: resources[1] };
+          });
+          data.spec.services = services;
+          project.createProject(data, () => {
+            history.replace(`/project`);
+          });
+        }}
+        render={props => (
```
Comment (caosm) [inline web/src/components/project/component/addProject/Form.js]: separate as an independent component

File: web/src/components/project/component/addProject/Form.js [hunk 5]
```diff
@@ -1,74 +1,111 @@
...
-        component={ResourceAllocation}
-        onChange={changeConfig}
+@inject('project')
+@observer
+class AddProject extends React.Component {
+  render() {
+    const { project, history } = this.props;
+    return (
+      <Formik
+        initialValues={{
```
Comment (caosm) [inline web/src/components/project/component/addProject/Form.js]: separate as a const

File: web/src/components/project/component/addProject/Form.js [hunk 6]
```diff
@@ -1,74 +1,111 @@
 import React from 'react';
 import { Form, Input, Button } from 'antd';
 import PropTypes from 'prop-types';
-import { Field, withFormik } from 'formik';
+import { Field, Formik } from 'formik';
+import { inject, observer } from 'mobx-react';
 import MakeField from '@/components/public/makeField';
 import IntegrationSelect from './IntegrationSelect';
-import ResourceAllocation from '@/components/public/resourceAllocation';
+import Quota from '@/components/public/quota';
```
Comment (caosm) [inline web/src/components/project/component/addProject/Form.js]: `Pascal` for component & component folder

File: web/src/components/project/component/addProject/Form.js [hunk 7]
```diff
@@ -1,74 +1,111 @@
...
+        }}
+        validate={values => {
+          const errors = {};
+
+          if (!values.metadata.name) {
+            errors.metadata = { name: 'Required' };
+          }
+          return errors;
+        }}
+        onSubmit={values => {
```
Comment (caosm) [inline web/src/components/project/component/addProject/Form.js]: separate

**Pull request merged**